MapleServer2/PacketHandlers/Game/UserChatHandler.cs

```diff
@@ -169,18 +194,82 @@ private static void HandleWhisperChat(GameSession session, string recipient, str
                 return;
             }
 
+            if (itemLinkPacket != null)
+            {
+                recipientPlayer.Session.Send(itemLinkPacket);
+                session.Send(itemLinkPacket);
+            }
             recipientPlayer.Session.Send(ChatPacket.Send(session.Player, message, ChatType.WhisperFrom));
             session.Send(ChatPacket.Send(recipientPlayer, message, ChatType.WhisperTo));
         }
 
-        private static void HandleClubChat(/*GameSession session, string message, ChatType type, long clubId*/)
+        private static void HandleClubChat(/*GameSession session, string message, ChatType type, long clubId, Packet itemLinkPacket*/)
         {
             // TODO
         }
 
-        private static void HandleChat(GameSession session, string message, ChatType type)
+        private static void HandleChat(GameSession session, string message, ChatType type, Packet itemLinkPacket)
         {
+            if (itemLinkPacket != null)
+            {
+                session.FieldManager.BroadcastPacket(itemLinkPacket);
+            }
             session.FieldManager.SendChat(session.Player, message, type);
         }
+
+        private static Packet GetItemLink(string message)
+        {
+            // '<' signals a message containing an item link
+            if (!message.Contains('<'))
+            {
+                return null;
+            }
+            Packet itemLinkPacket = null;
+
+            Match a = Regex.Match(message, @"<A (.*?)>");
+            string[] itemLinkMessages = Regex.Matches(message, @"<A (.*?)>")
+                .Cast<Match>()
+                .Select(m => m.Value)
+                .ToArray();
+            List<Item> items = new();
+
+            foreach (string itemLinkMessage in itemLinkMessages)
+            {
+                string[] itemLinkMessageSplit = itemLinkMessage.Split(',');
+                string itemLinkType = Regex.Match(itemLinkMessageSplit[0], @"(?<=:)(.*)").Value;
+
+                long itemUid = long.Parse(itemLinkMessageSplit[1]);
+                Item item = null;
+                switch (itemLinkType)
+                {
+                    case "itemTooltip":
```

Comment:
switch with only one case? It's supposed to be more cases? If so, add an comment